src/request/component-models/ComponentModel.ts

```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+import { IComponentModel } from './IComponentModel';
+
+export abstract class ComponentModel implements IComponentModel {
+    public abstract init(request);
+}
```

Comment:
Since it was already there, I kept it "just in case" there was a future need to have some base functionality for each component model.